node/src/components/storage.rs

```diff
@@ -534,6 +533,7 @@ impl Storage {
                     block.header().hash(),
                     block.body(),
                 )?;
```

Comment:
You're not passing in `txn`, so I don't think this makes sure they actually use the same transaction?